staff/lila-petri/array-functions/filter.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Iterates an array and get a new array form the original array
+ * @param {Array} array 
+ * @param {Function} callback The expression to evaluate.
+ */
+
+function filter(array, callback) {
+    if (!(array instanceof Array)) throw TypeError(array + ' is not an array');
+    if (typeof callback !== 'function') throw new TypeError(callback + ' is not a function');
+    var newarray=[];
+    for (var i = 0; i < array.length; i++){
+        if(callback(array[i], i, array)){
+            newarray[i]=array[i]
```

Comment:
if you do this way you will return array with holes.
function even (v) { return v % 2 === 0};
filter([4,6,9,10,5, 20,], even}
// [4,6,empty,10, empty,20]

Reply:
I'm not sure if I've fixed the problem properly. Can we talk about it?